internal/described_errors/described_errors.go

```diff
@@ -76,11 +77,18 @@ func (d DefaultDescribedError) ShouldSetCondition() bool {
 func GetMostSevereErr(errs []DescribedError) DescribedError {
 	var candidate DescribedError
 	for _, err := range errs {
-		// This checks if the current error has a lower level than the candidate. This might be counterintuitive, but it's correct because
-		// levels are ordered from the most severe to the least severe. So, if the current error has a lower level than the candidate, it's more severe.
-		if candidate == nil || err.Level() < candidate.Level() {
+		if err == nil {
+			continue
+		}
+		if candidate == nil || err.hasHigherSeverity(err) {
```

Comment:
Units definitely should have failed here
```suggestion
		if candidate == nil || candidate.hasHigherSeverity(err) {
```